sdk/samples/iot/iot_sample_common.h

```diff
@@ -18,65 +18,61 @@
 //
 // Logging
 //
-#define IOT_SAMPLE_LOG_ERROR(...) \
-  do \
-  { \
+#define IOT_SAMPLE_LOG_ERROR(...)                                                  \
+  do                                                                               \
+  {                                                                                \
     (void)fprintf(stderr, "ERROR:\t\t%s:%s():%d: ", __FILE__, __func__, __LINE__); \
-    (void)fprintf(stderr, __VA_ARGS__); \
-    (void)fprintf(stderr, "\n"); \
-    fflush(stdout); \
-    fflush(stderr); \
+    (void)fprintf(stderr, __VA_ARGS__);                                            \
+    (void)fprintf(stderr, "\n");                                                   \
+    fflush(stdout);                                                                \
+    fflush(stderr);                                                                \
   } while (0)
 
 #define IOT_SAMPLE_LOG_SUCCESS(...) \
-  do \
-  { \
-    (void)printf("SUCCESS:\t"); \
-    (void)printf(__VA_ARGS__); \
-    (void)printf("\n"); \
+  do                                \
+  {                                 \
+    (void)printf("SUCCESS:\t");     \
+    (void)printf(__VA_ARGS__);      \
+    (void)printf("\n");             \
   } while (0)
 
-#define IOT_SAMPLE_LOG(...) \
-  do \
-  { \
-    (void)printf("\t\t"); \
+#define IOT_SAMPLE_LOG(...)    \
+  do                           \
+  {                            \
+    (void)printf("\t\t");      \
     (void)printf(__VA_ARGS__); \
-    (void)printf("\n"); \
+    (void)printf("\n");        \
   } while (0)
 
-#define IOT_SAMPLE_LOG_AZ_SPAN(span_description, span) \
-  do \
-  { \
-    (void)printf("\t\t%s ", span_description); \
-    char* buffer = (char*)az_span_ptr(span); \
-    for (int32_t az_span_i = 0; az_span_i < az_span_size(span); az_span_i++) \
-    { \
-      putchar(*buffer++); \
-    } \
-    (void)printf("\n"); \
+#define IOT_SAMPLE_LOG_AZ_SPAN(span_description, span)                                           \
+  do                                                                                             \
+  {                                                                                              \
+    (void)printf("\t\t%s ", span_description);                                                   \
+    (void)fwrite((char*)az_span_ptr(span), sizeof(uint8_t), (size_t)az_span_size(span), stdout); \

```

Comment:
Seems reasonable. The only nit I have is that some IoT devices don't have a filesystem, so using `fwrite` in samples (which is primarily for files, even though it is writing to stdout), might give the wrong impression to customers.